gguf-py/gguf/gguf_writer.py

```diff
@@ -417,6 +419,19 @@ def write_tensor_data(self, tensor: np.ndarray[Any, Any]) -> None:
 
         self.state = WriterState.WEIGHTS
 
+    def generate_tensors_uuid(self) -> str:
+        uuidv5_sha1 = hashlib.sha1()
+        uuidv5_sha1.update(uuid.UUID('ef001206-dadc-5f6d-a15f-3359e577d4e5').bytes)
+
+        for tensors in self.tensors:
+            # relying on the fact that Python dicts preserve insertion order (since 3.7)
+            for name, ti in tensors.items():
+                assert ti.tensor is not None
+                assert ti.tensor.nbytes == ti.nbytes
+                uuidv5_sha1.update(ti.tensor.tobytes('C'))
```

Comment:
While writing <https://github.com/ggerganov/llama.cpp/pull/8645#issuecomment-2246138344>, I've realized that this specific line is materializing the lazy tensors by reading their data, which would cause a very noticeable memory regression (making it no better than `--no-lazy`, which is not good RAM-usage-wise), at least when there is no UUID specified (which means, by default).

If you can prove this is false, I'd be happy to dismiss this review.